### ISSUE TITLE
Code Quality: Convert into `await using` statement or declaration

### ISSUE THREAD

**Issue by gumbarros** [files-community/Files#15092]
### Description

```
using (var stream = File.OpenRead(_item.ItemPath))
```
to

```
await using (var stream = File.OpenRead(_item.ItemPath))					
```

In .NET, classes that own unmanaged resources usually implement the [IDisposable](https://learn.microsoft.com/en-us/dotnet/api/system.idisposable) interface to provide a mechanism for releasing unmanaged resources synchronously. However, in some cases they need to provide an asynchronous mechanism for releasing unmanaged resources in addition to (or instead of) the synchronous one. Providing such a mechanism enables the consumer to perform resource-intensive dispose operations without blocking the main thread of a GUI application for a long time.

The [IAsyncDisposable.DisposeAsync](https://learn.microsoft.com/en-us/dotnet/api/system.iasyncdisposable.disposeasync) method of this interface returns a [ValueTask](https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.valuetask) that represents the asynchronous dispose operation. Classes that own unmanaged resources implement this method, and the consumer of these classes calls this method on an object when it is no longer needed.

Source: https://stackoverflow.com/questions/58610350/what-is-the-difference-between-using-and-await-using-and-how-can-i-decide-which

### Concerned code

Generated using Jetbrains Rider:

<src>\<platforms>\<Files.App>\Utils\Storage\StorageItems\VirtualStorageItem.cs (1 issue)
<src>\<platforms>\<Files.App>\Utils\Storage\StorageBaseItems\BaseStorageFile.cs (1 issue)
<src>\<platforms>\<Files.App>\Utils\Shell\LaunchHelper.cs (2 issues)
<src>\<platforms>\<Files.App>\Utils\Storage\StorageItems\ZipStorageFile.cs (7 issues)
<src>\<platforms>\<Files.App>\Utils\Storage\StorageItems\VirtualStorageFile.cs (4 issues)
<src>\<platforms>\<Files.App>\Utils\Storage\Operations\ShellFilesystemOperations.cs (4 issues)
<src>\<platforms>\<Files.App>\Utils\Storage\StorageItems\ZipStorageFolder.cs (9 issues)
<src>\<platforms>\<Files.App>\Utils\Storage\StorageItems\NativeStorageFile.cs (4 issues)
<src>\<platforms>\<Files.App>\Utils\Storage\StorageItems\SystemStorageFile.cs (9 issues)
<src>\<platforms>\<Files.App>\ViewModels\Properties\HashesViewModel.cs (1 issue)
<src>\<platforms>\<Files.App>\Utils\Storage\StorageItems\FtpStorageFile.cs (3 issues)
<src>\<platforms>\<Files.App>\Utils\Cloud\Detector\SynologyDriveCloudDetector.cs (3 issues)
<src>\<platforms>\<Files.App>\Utils\Cloud\Detector\GoogleDriveCloudDetector.cs (3 issues)
<src>\<platforms>\<Files.App>\Services\SideloadUpdateService.cs (2 issues)
<src>\<platforms>\<Files.App>\Services\UpdateService.cs (2 issues)
<src>\<platforms>\<Files.App>\Extensions\ImageSourceExtensions.cs (1 issue)

### Gains

- UI will not freeze while disposing a resource

### Requirements

- Convert all asynchronous using to `async using`

### Comments

![image](https://github.com/files-community/Files/assets/52143624/4ac9f73c-f404-4a83-b402-bc655646e954)

It's possible to make a single PR using this button :)

**Comment by 0x5bfa** [comment]
Out of curious, what IDE are you using?

**Comment by gumbarros** [comment]
> Out of curious, what IDE are you using? I also wanna generate this list lol.

Rider, but you can install ReSharper at VS to obtain the same result.

**Comment by 0x5bfa** [comment]
Oh they are kind. Why do you know those warnings won't be shown in VS2022 by default and what should we do to should those issues that would improve performance.

**Comment by 0x5bfa** [comment]
@yaira2 Can I use the name of Files Community for:
![image](https://github.com/files-community/Files/assets/62196528/79a0c5c8-a87e-4e61-9466-b1504941b894)

**Comment by gumbarros** [comment]
> Oh they are kind. Why do you think those warnings won't be shown in VS2022 by default and what should we do to should those issues that would improve performance.

I guess this is why JetBrains makes money lol, if VS showed these problems ReSharper wouldn't exist. Maybe because of some limitation?

Yair said about benchmarking, I think this is valid. There is also dozens of issues types when I run the analyzer, but I opened just the most importants, some of them I think they don't make sense, like `var` instead of the explicit type, not worth a PR.

![image](https://github.com/files-community/Files/assets/52143624/58fd2ade-7719-4de7-ab43-278a5574f334)

**Comment by mafra99** [comment]
I'm not sure if Files counts as a project, since users have to pay for the version in the Microsoft Store

From here https://www.jetbrains.com/community/opensource/#support

![image](https://github.com/files-community/Files/assets/56681014/ae09f7d0-a6b3-4cd4-b817-257f34ae712c)

**Comment by 0x5bfa** [comment]
That's absolutely right lol I was wondering why VS doesn't show warnings that comes from the official analyzers(IDE????, C????) while Rider shows. But I seemed to have misunderstood. Thanks.

Really nice features.

**Comment by gumbarros** [comment]
> I'm not sure if Files counts as a project, since users have to pay for the version in the Microsoft Store
> 
> From here https://www.jetbrains.com/community/opensource/#support
> 
> ![image](https://private-user-images.githubusercontent.com/56681014/318447637-ae09f7d0-a6b3-4cd4-b817-257f34ae712c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE5NzkwNTIsIm5iZiI6MTcxMTk3ODc1MiwicGF0aCI6Ii81NjY4MTAxNC8zMTg0NDc2MzctYWUwOWY3ZDAtYTZiMy00Y2Q0LWI4MTctMjU3ZjM0YWU3MTJjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDAxVDEzMzkxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTYwY2VmY2MxM2Y3MWZkMjAxYTEyMWNmNmEwMDU3NjEwYTkxMmI4YWZhMzYxZDM0M2Q2ZDViM2VhMDMyODIzMTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.6Qw4FQooYZhfDW4R7_gEqiS7jCwqY0f4QVbG4hsk6js)

![image](https://github.com/files-community/Files/assets/52143624/cb4d3a10-586d-450a-89fa-8e48f23c0e39)
I saw some projects with "Thanks for JetBrains" at the end of the README file, maybe if you contact JetBrains sales they can make a exception for Files as this is a big project .

**Comment by 0x5bfa** [comment]
But that's for donation.

**Comment by yaira2** [comment]
> I'm not sure if Files counts as a project, since users have to pay for the version in the Microsoft Store

I wouldn't consider the store version as commercial, the sideload version has all the same features and users get the same level of support. In fact, the sideload version has many times the number of users as the store version.

**Comment by 0x5bfa** [comment]
Sorry for the unrelated comments, Yair. I don't mind you deleting them.

Besides that, I'm onboard where applicable. Basically classes that implement IAsyncDisposable need to release unmanaged objects asynchronously but using Disposable may cause hangs.
